plutus-core/plutus-core/src/PlutusCore/Subst.hs

```diff
@@ -161,6 +164,56 @@ termSubstClosedTerm varFor new = go where
          t                    -> t & over termSubterms go
     goUnder var term = if var == varFor then term else go term
 
+-- Mapping names
+
+typeMapNames
+    :: forall tyname tyname' uni ann
+    .  (tyname -> tyname')
+    -> Type tyname uni ann
+    -> Type tyname' uni ann
+typeMapNames g = go
+    where
+      go :: Type tyname uni ann -> Type tyname' uni ann
+      go = \case
+           TyVar ann tn         -> TyVar ann (g tn)
+           TyFun ann ty1 ty2    -> TyFun ann (go ty1) (go ty2)
+           TyIFix ann ty1 ty2   -> TyIFix ann (go ty1) (go ty2)
+           TyForall ann tn k ty -> TyForall ann (g tn) k (go ty)
+           TyBuiltin ann s      -> TyBuiltin ann s
+           TyLam ann tn k ty    -> TyLam ann (g tn) k (go ty)
+           TyApp ann ty1 ty2    -> TyApp ann (go ty1) (go ty2)
+
+-- termMapNames requires two function arguments: one (called g) to map type names
+-- and another (called f) to map variable names.
+termMapNames
+    :: forall tyname tyname' name name' uni fun ann
+    .  (tyname -> tyname')
+    -> (name -> name')
+    -> Term tyname name uni fun ann
+    -> Term tyname' name' uni fun ann
+termMapNames g f = go
+    where
```

Comment:
Nit: same for this g

Reply:
Another idea that would perhaps fit better is that `termMapNames` accepts one function only (name -> name') and leaves tynames untouched.

Then have another function `termMapBothNames = typeMapNames . termMapNames`. This ofcourse is slower, but is more consistent. Right now termMapNames also maps types

Reply:
Yes, that might be better. All of this would be a lot easier if TPLC just had one name parameter that it used for both types and terms.

Reply:
Oh wait, that doesn't quite work.  `typeMapNames` just maps a function over a type, so we'd need another full traversal over terms that applies `typeMapNames` to the types in the term.  I think I'll leave that as it is for now.